sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeActionsAsyncClient.java

```diff
@@ -658,86 +960,244 @@ private AnalyzeActionsResult toAnalyzeActionsResult(AnalyzeJobState analyzeJobSt
+                    analyzeHealthcareEntitiesActionResults.add(actionResult);
+                } else if (taskResult instanceof SentimentLROResult) {
+                    final SentimentLROResult sentimentLROResult = (SentimentLROResult) taskResult;
+                    final AnalyzeSentimentActionResult actionResult = new AnalyzeSentimentActionResult();
+                    final SentimentResponse results = sentimentLROResult.getResults();
+                    if (results != null) {
+                        AnalyzeSentimentActionResultPropertiesHelper.setDocumentsResults(actionResult,
+                            toAnalyzeSentimentResultCollection(results));
+                    }
+                    TextAnalyticsActionResultPropertiesHelper.setActionName(actionResult,
+                        sentimentLROResult.getTaskName());
+                    TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
+                        sentimentLROResult.getLastUpdateDateTime());
+                    analyzeSentimentActionResults.add(actionResult);
+                } else if (taskResult instanceof KeyPhraseExtractionLROResult) {
+                    final KeyPhraseExtractionLROResult keyPhraseExtractionLROResult =
+                        (KeyPhraseExtractionLROResult) taskResult;
+                    final ExtractKeyPhrasesActionResult actionResult = new ExtractKeyPhrasesActionResult();
+                    final KeyPhraseResult results = keyPhraseExtractionLROResult.getResults();
+                    if (results != null) {
+                        ExtractKeyPhrasesActionResultPropertiesHelper.setDocumentsResults(actionResult,
+                            toExtractKeyPhrasesResultCollection(results));
+                    }
+                    TextAnalyticsActionResultPropertiesHelper.setActionName(actionResult,
+                        keyPhraseExtractionLROResult.getTaskName());
+                    TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
+                        keyPhraseExtractionLROResult.getLastUpdateDateTime());
+                    extractKeyPhrasesActionResults.add(actionResult);
+                } else {
+                    throw logger.logExceptionAsError(new RuntimeException(
+                        "Invalid Long running operation task result: " + taskResult.getClass()));
                 }
-                TextAnalyticsActionResultPropertiesHelper.setActionName(actionResult, taskItem.getTaskName());
-                TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
-                    taskItem.getLastUpdateDateTime());
-                multiCategoryClassifyActionResults.add(actionResult);
             }
         }
 
-        final List<TextAnalyticsError> errors = analyzeJobState.getErrors();
+        //TODO: In Language REST API, there is no such task names. It might be a different way to parse the Error Target
```

Comment:
what's the behavior in the java library if a target is not returned by the service?

Reply:
Java will throw an error

  if (CoreUtils.isNullOrEmpty(targetReference)) {
            throw logger.logExceptionAsError(new RuntimeException(
                "Expected an error with a target field referencing an action but did not get one"));
        }

Reply:
Would it make sense to throw the error message(s) that the service returns in the error object? While the error you throw now is accurate, it kind of focuses more on the internal implementation.

Reply:
updated